ledger-service/http-json/src/main/scala/com/digitalasset/http/Main.scala

```diff
@@ -4,46 +4,51 @@
 package com.daml.http
 
 import java.nio.file.Path
-
 import akka.actor.ActorSystem
 import akka.http.scaladsl.Http.ServerBinding
 import akka.stream.Materializer
 import com.daml.grpc.adapter.{AkkaExecutionSequencerPool, ExecutionSequencerFactory}
 import com.daml.runtime.JdbcDrivers
 import com.daml.scalautil.Statement.discard
 import com.daml.http.dbbackend.ContractDao
-import com.typesafe.scalalogging.StrictLogging
 import scalaz.{-\/, \/, \/-}
 import scalaz.std.anyVal._
 import scalaz.std.option._
 import scalaz.syntax.show._
 import com.daml.cliopts.GlobalLogLevel
+import com.daml.http.util.Logging.{CorrelationID, correlationIdLogCtx, extendWithRequestIdLogCtx}
+import com.daml.logging.{ContextualizedLogger, LoggingContextOf}
 
 import scala.concurrent.duration._
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-object Main extends StrictLogging {
+object Main {
+
+  private[this] def logger = ContextualizedLogger.get(getClass)
 
   object ErrorCodes {
     val Ok = 0
     val InvalidUsage = 100
     val StartupError = 101
   }
 
-  def main(args: Array[String]): Unit =
-    Cli.parseConfig(
-      args,
-      ContractDao.supportedJdbcDriverNames(JdbcDrivers.availableJdbcDriverNames),
-    ) match {
-      case Some(config) =>
-        main(config)
-      case None =>
-        // error is printed out by scopt
-        sys.exit(ErrorCodes.InvalidUsage)
-    }
+  def main(args: Array[String]): Unit = {
+    correlationIdLogCtx(implicit lc =>
```

Comment:
What is exactly the correlation identifier? Does it identify exactly an individual HTTP JSON API server process?

Reply:
Yes it should, it's not using the process id but a randomly generated UUID which I hope fulfills the requirement in 99% of the time.